src/controller/java/src/chip/onboardingpayload/ManualOnboardingPayloadGenerator.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package chip.onboardingpayload
+
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class ManualOnboardingPayloadGenerator(private val payloadContents: OnboardingPayload) {
+  private var allowInvalidPayload = false
+  private var forceShortCode = false
+
+  fun setAllowInvalidPayload(allow: Boolean) {
```

Comment:
skipPayloadValidation or similar seems a better name.

A `allow invalid payloads` does sound very odd. In fact this seems to only control invalid manual codes. I would prefer to narrow down the naming to `skipManualCodeValidationCheck` or similar.